tests/test_layout.py

```diff
@@ -12,8 +12,12 @@ class TestLayout:
     @pytest.mark.nondestructive
     @pytest.mark.xfail("'mozilla.com' in config.getvalue('base_url')",
                        reason="https://bugzilla.mozilla.org/show_bug.cgi?id=1201622")
-    def test_that_products_are_sorted_correctly(self, mozwebqa):
-        csp = CrashStatsHomePage(mozwebqa)
+    def test_that_products_are_sorted_correctly(self, base_url, selenium):
+        # TODO what is the proper way to do a conditional xfail? IIRC this
+        # syntax will prevent the testcase from actually running
+        # if 'mozilla.com' in base_url:
+        #     pytest.mark.xfail(reason="https://bugzilla.mozilla.org/show_bug.cgi?id=1201622")
```

Comment:
The original should work here, although it doesn't need to be a string in latest versions of py.test:

```python
 @pytest.mark.xfail('mozilla.com' in pytest.config.getvalue('base_url'),
                    reason='https://bugzilla.mozilla.org/show_bug.cgi?id=1201622')
```

That said, if the base URL is provided via configuration file such as `setup.cfg` this will not work. It's possible that `config.getini('base_url')` would cover that case, but we'd still have a problem if the base URL is provided by a fixture. For purposes of our Jenkins jobs, I think I'm fine with assuming the base URL is provided by command line.

In a future version of pytest-selenium we may try to provide a consistent way to access the base URL regardless of the way it's defined. This still won't work for fixture as they're executed at runtime and xfailing is run at collection time, but it will at least cover configuration files and command line options.

I'm interested to hear any other creative suggestions for how to xfail based on target environments.